packages/shared/scope-hoisting/src/generate.js

```diff
@@ -109,6 +110,33 @@ export function generate({
   if (bundle.env.sourceMap && rawMappings != null) {
     map = new SourceMap(options.projectRoot);
     map.addIndexedMappings(rawMappings);
+
+    // Traverse the bundle to get the sourcecontents
+    // this is hella slow but is currently the only way to ensure correct source contents
+    let promiseQueue = new PromiseQueue({maxConcurrent: 50});
+    bundle.traverseAssets(asset => {
+      promiseQueue.add(async () => {
+        // Why is map always undefined?
+        let map = await asset.getMap();
```

Comment:
These maps don't get set, because generate never gets called.

However for runtimes these maps already exist before `generate()` so they exist, but are not part of the asset here for some reason...

However scope-hoisting should already have these maps embedded in the AST (which it does not because babel parser does not support input maps at the moment), but even if they did the sourcecontents would be missing?

There's a couple solutions to this:
- Save sourcecontents seperately in the asset (this will probably duplicate it as the map also contains it...)
- Add support for input sourcemaps and ideally also for sourcecontents to babel/parser
- Allow the scope-hoisting bundler from accessing the runtime maps.

In the future we probably also want compiled/bundled node_modules to map back to their original source, for the babel would ideally support inputMap in the parser (https://github.com/babel/babel/issues/11960)